Week3/homework/step2-6.js

```diff
@@ -4,19 +4,20 @@ const arr2d = [[1, 2], [3, 4], [5, 6]];
 const arr3d = [[[1, 2], [3, 4]], [[5, 6], [7, 8]]];
 
 function flattenArray2d(arr) {
-  // Replace this comment and the next line with your code
-  console.log(arr);
+  while (arr.find(item => typeof item === 'object') !== undefined) arr = [].concat(...arr);
```

Comment:
While your code is clever and passes the test, the solution that you have chosen is fragile. It works for the array provided, but if I add an object literal as an array element your code loops forever:

```js
const arr2d = [[1, 2], [3, 4], [5, 6], {}];
```

This is because an array is an object, for sure, but so is an actual object. If you want to test whether a value is an array (and not just any object) you should use `Array.isArray()`:

```js
while (arr.find(item => Array.isArray(item))) arr = [].concat(...arr);
```